File: src/dataset/cpp/raindrops_exercism-stungeye/openAI/ChatGPT4_raindrops_v3.cpp
#include "raindrops.h"

namespace raindrops {

	std::string convert(int number) {
		bool hasFactor = false;
		std::string raindrops;

		if (number % 3 == 0) {
			raindrops = "Pling";
			hasFactor = true;
		}

		if (number % 5 == 0) {
			raindrops += "Plang";
			hasFactor = true;
		}

		if (number % 7 == 0) {
			raindrops += "Plong";
			hasFactor = true;
		}

		return hasFactor ? raindrops : std::to_string(number);
	}
} // namespace raindrops

File: src/dataset/typescript/raindrops_exercism-wackerow/openAI/ChatGPT4_raindrops_v3.ts
export default class Raindrops {
  public convert(num: number): string {
    const mappings = [
      [3, 'Pling'],
      [5, 'Plang'],
      [7, 'Plong']
    ] as const;

    let raindrop = mappings.reduce((acc, [factor, sound]) => 
      num % factor === 0 ? acc + sound : acc, '');

    return raindrop || num.toString();
  }
}